src/main/arm/createUiDefinition.json

```diff
@@ -245,6 +245,27 @@
                         },
                         "visible": "[bool(basics('basicsOptional').basicsOptionalAcceptDefaults)]"
                     },
+                    {
+                        "name": "useSystemAssignedManagedIdentity",
+                        "label": "Cause a system assigned managed identity to be created for the VM(s).",
+                        "type": "Microsoft.Common.OptionsGroup",
+                        "toolTip": "System assigned managed identities enable credential-free secure access to many Azure resources from this VM.",
+                        "constraints": {
+                            "allowedValues": [
+                                {
+                                    "label": "Yes",
+                                    "value": true
+                                },
+                                {
+                                    "label": "No",
+                                    "value": false
+                                }
+                            ],
+                            "required": true
+                        },
+                        "defaultValue": "Yes",
+                        "visible": "[bool(basics('basicsOptional').basicsOptionalAcceptDefaults)]"
+                    }, 

```

Comment:
I am not sure why you did not use the same configuration I offered in my branch on `arm-oraclelinux-wls`:

```json
                    {
                        "name": "useSystemAssignedManagedIdentity",
                        "label": "Cause a system assigned managed identity to be created for the VM(s).",
                        "type": "Microsoft.Common.CheckBox",
                        "toolTip": "System assigned managed identities enable credential-free secure access to many Azure resources from this VM.",
                        "defaultValue": true
                    },            
...
            "useSystemAssignedManagedIdentity": "[basics('basicsOptional').useSystemAssignedManagedIdentity]",        
```

A checkbox is a better UI element than Yes/No for this use case.